actionpack/lib/action_dispatch/routing.rb

```diff
@@ -121,8 +121,8 @@ module ActionDispatch
   #     get 'blog/edit/:id' => :edit
   #   end
   #
-  #   # provides named routes for show, delete, and edit
-  #   link_to @article.title, show_path(id: @article.id)
+  #   # provides named routes for show, delete
```

Comment:
Actually, can you fix and remove `/:id` from edit route, then it will generate all three named routes.